### PR TITLE
cortex/redisclient: use rueidis client

### LINKED ISSUE #6094: [query-frontend] Unable to connect to Redis 6.0
**Thanos, Prometheus and Golang version used**: Thanos 0.30.1, Azure Cache for Redis 6.0

**Object Storage Provider**: Azure

**What happened**: Attempting to configure result and label caches for the query-frontend. The redis cache is not reachable, context times out. The same configuration works for the store gateway component.

**What you expected to happen**: Query frontend can connect to Redis.

**How to reproduce it (as minimally and precisely as possible)**: 
- Deploy query frontend version 0.30.1 with the following cache config:
```
type: REDIS
config:
  password: <redacted>
  addr: <account>.redis.cache.windows.net:6380
  cache_size: 4GB
  tls_enabled: true
```

**Full logs to relevant components**:
```
level=info ts=2023-02-01T08:09:07.755438545Z caller=factory.go:43 msg="loading tracing configuration"
level=error ts=2023-02-01T08:09:10.764288508Z caller=redis_cache.go:46 msg="error connecting to redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:09:13.765598973Z caller=redis_cache.go:46 msg="error connecting to redis" name=redis err="context deadline exceeded"
level=info ts=2023-02-01T08:09:13.766096073Z caller=query_frontend.go:339 msg="starting query frontend"
level=info ts=2023-02-01T08:09:13.766118273Z caller=intrumentation.go:56 msg="changing probe status" status=ready
level=info ts=2023-02-01T08:09:13.766175773Z caller=intrumentation.go:75 msg="changing probe status" status=healthy
level=info ts=2023-02-01T08:09:13.766202673Z caller=http.go:73 service=http/server component=query-frontend msg="listening for requests and metrics" address=0.0.0.0:9090
level=info ts=2023-02-01T08:09:13.766349073Z caller=tls_config.go:232 service=http/server component=query-frontend msg="Listening on" address=[::]:9090
level=info ts=2023-02-01T08:09:13.766373573Z caller=tls_config.go:235 service=http/server component=query-frontend msg="TLS is disabled." http2=false address=[::]:9090
level=error ts=2023-02-01T08:11:46.611441925Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611464125Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611490825Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611513125Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611530625Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611617825Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611641925Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611717524Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611729224Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611732624Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611743124Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611795124Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611812024Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611825924Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611844124Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611505025Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611871924Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
level=error ts=2023-02-01T08:11:46.611873224Z caller=redis_cache.go:75 msg="failed to get from redis" name=redis err="context deadline exceeded"
```

**Anything else we need to know**:
I see https://github.com/thanos-io/thanos/issues/6018 but don't think this is related as we are using Redis 6, not 7.

### PR DESCRIPTION
**Pull request by GiedriusS**
Use the same rueidis client in query-frontend. Solves https://github.com/thanos-io/thanos/issues/6094.

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Use rueidis client instead of go-redis in query-frontend to fix #6094.

## Verification

Tests.

### CODE REVIEW

File: internal/cortex/chunk/cache/cache.go [hunk 1]
```diff
@@ -101,7 +102,11 @@ func New(cfg Config, reg prometheus.Registerer, logger log.Logger) (Cache, error
 			cfg.Redis.Expiration = cfg.DefaultValidity
 		}
 		cacheName := cfg.Prefix + "redis"
-		cache := NewRedisCache(cacheName, NewRedisClient(&cfg.Redis), reg, logger)
+		redisClient, err := NewRedisClient(&cfg.Redis)
+		if err != nil {
+			return nil, fmt.Errorf("creating redis client: %w", err)
```
[the remark this answers is not in the document]
Comment (GiedriusS) [reply internal/cortex/chunk/cache/cache.go:107]: :+1: good point, updated

File: internal/cortex/chunk/cache/cache.go [hunk 2]
```diff
@@ -5,10 +5,11 @@ package cache
 
 import (
 	"context"
-	"errors"
 	"flag"
 	"time"
 
+	"github.com/pkg/errors"
+
```
Comment (squat) [inline internal/cortex/chunk/cache/cache.go:12]: Nit: this seems like an extra newline

**Review by squat — APPROVED** [review]
Nice work :)))

File: internal/cortex/chunk/cache/cache.go (hunk 2 above)
Comment (GiedriusS) [reply internal/cortex/chunk/cache/cache.go:12]: Actually not, `gofumptd` and `go fmt` adds this :smile:

**Review by fpetkovski — APPROVED** [review]

**Review by squat — APPROVED** [review]

**Pull request merged**